File: Beecrowd/1256 Tabelas Hash/Main.java
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();

        for (int i = 0; i < n; i++) {
            int m = sc.nextInt();
            int c = sc.nextInt();

            HashTable tab = new HashTable(m);
            StringBuilder sb = new StringBuilder();

            for (int j = 0; j < c; j++) {
                int valor = sc.nextInt();
                tab.inserirValor(valor);
            }

            tab.imprimirHash(sb);
            if (i < n - 1)
                System.out.println(sb);
            else
                System.out.print(sb);
        }

        sc.close();
    }
}

class HashTable {

    private final List<LinkedList<Integer>> tab;
    private int tam;
    private int pos;

    public HashTable(int tam) {
        this.tam = tam;
        this.tab = new ArrayList<>(tam);
        inicializaLista();
    }

    private void inicializaLista() {
        for (int i = 0; i < tam; i++) {
            this.tab.add(new LinkedList<>());
        }
    }

    private int funcaoHash(int valor) {
        return valor % tam;
    }

    public void inserirValor(int valor) {
        this.pos = funcaoHash(valor);
        LinkedList<Integer> list = tab.get(pos);
        list.add(valor);
    }

    public void imprimirHash(StringBuilder sb) {
        for (int i = 0; i < tab.size(); i++) {
            sb.append(i + " -> ");
            LinkedList<Integer> hash = tab.get(i);

            if (!hash.isEmpty()) {
                for (int j = 0; j < hash.size(); j++) {
                    if (j < hash.size() - 1)
                        sb.append(hash.get(j) + " -> ");
                    else
                        sb.append(hash.get(j) + " -> \\");
                }
                sb.append("\n");
            } else
                sb.append("\\\n");
        }
    }
}

File: Beecrowd/1256 Tabelas Hash/main.py
class HashTable:
    def __init__(self, tam):
        self.tam = tam
        self.tab = [[] for _ in range(tam)]

    def funcao_hash(self, valor):
        return valor % self.tam
    
    def inserir_valor(self, valor):
        self.pos = self.funcao_hash(valor)
        list = self.tab[self.pos]
        list.append(valor)
    
    def imprimir_hash(self):
        for i, h in enumerate(self.tab):
            print(f"{i} -> ", end="")
            if not h:
                print("\\")
            else:
                for j, valor in enumerate(h):
                    if j < len(h) - 1:
                        print(f"{valor} -> ", end="")
                    else:
                        print(f"{valor} -> \\", end="")
                print()

if __name__ == '__main__':
    n = int(input())

    for i in range(n):
        entrada = [int(_) for _ in input().split()]
        m = entrada[0]
        c = entrada[1]

        tab = HashTable(m)

        valores = [int(_) for _ in input().split()[:c]]

        for j in range(c):
            tab.inserir_valor(valores[j])

        if(i > 0):
            print()

        tab.imprimir_hash()
